### ISSUE TITLE
Turning on debug server causes infinite excel launch loop

### ISSUE THREAD

**Issue by airguru** [cunnane/xloil#20]
Turning on the debugger option in ribbon causes the start of a new excel instance. Then, a series of these messages follow:
![image](https://user-images.githubusercontent.com/728780/206307194-eabd8742-7b19-4857-b730-7b2b8424c6c4.png)
![image](https://user-images.githubusercontent.com/728780/206307301-3e3372f1-2dd1-4fe0-a5ac-78bf60da3e84.png)
![image](https://user-images.githubusercontent.com/728780/206307355-a5990eec-e6fc-4f84-89ab-2e982fcca934.png)
![image](https://user-images.githubusercontent.com/728780/206307422-8739c179-b50f-4ea7-9b1a-4827b6e91ee5.png)
![image](https://user-images.githubusercontent.com/728780/206307467-0addd9f9-e80b-4548-ae4d-969d202143d4.png)

After dismissing them, another instance of excel opens with the same message (presumably because the debugger is enabled there as well). This continues indefinitely until all excel processes are killed and debug server is disabled in the .ini file.

Clearly this looks like some command-line problem, where the debug server arguments are passed to excel executable.

Meanwhile, console shows these errors:
```

[23:08:26] [error] Error during coroutine: Traceback (most recent call last):
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\debugpy\server\api.py", line 237, in listen
    sock, _ = endpoints_listener.accept()
  File "c:\users\airgu\appdata\local\programs\python\python37\Lib\socket.py", line 212, in accept
    fd, addr = self._accept()
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\xloil\register.py", line 139, in _logged_wrapper_async
    return await coro
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\xloil\xloil_ribbon.py", line 266, in set_debugger
    port=int(_settings.python['DebugPyPort']))
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\xloil\debug.py", line 163, in use_debugger
    debugpy_listen(**kwargs)
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\xloil\debug.py", line 60, in debugpy_listen
    connection = debugpy.listen(port)
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\debugpy\public_api.py", line 31, in wrapper
    return wrapped(*args, **kwargs)
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\debugpy\server\api.py", line 143, in debug
    log.reraise_exception("{0}() failed:", func.__name__, level="info")
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\debugpy\server\api.py", line 141, in debug
    return func(address, settrace_kwargs, **kwargs)
  File "c:\users\airgu\appdata\local\programs\python\python37\lib\site-packages\debugpy\server\api.py", line 251, in listen
    raise RuntimeError("timed out waiting for adapter to connect")
RuntimeError: timed out waiting for adapter to connect
```

I do have Python 3.7, `debugpy` package is installed. Windows 11, Excel 2010.

**Comment by cunnane** [comment]
You're exactly correct, *debugpy* assumes that *sys.executable* is set to something ending in `python.exe` which isn't the default case when python is embedded.  xlOil works around this by setting the path explicitly in the interpreter's pre-load config.  However due to differences in interpreter initialisation, this fix was only added for python 3.8+.  I'll figure out what the equivalent setting is for older versions and make a patch.

**Comment by cunnane** [comment]
I've put a fix for this in v0.15.4 which is available now.

**Comment by airguru** [comment]
Thank you very much. Unfortunately I have broken my python environment in the meantime so I reinstalled straight into new version of Python. So I cannot test it :(
